qiskit/primitives/base_estimator.py

```diff
@@ -0,0 +1,236 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+r"""
+=========
+Estimator
+=========
+
+Estimator class estimates expectation values of quantum circuits and observables.
+
+An estimator object is initialized with multiple quantum circuits and observables
+and users can specify pairs of quantum circuits and observables
+to estimate the expectation values.
+
+The input consists of following elements.
+
+* quantum circuits (:math:`\psi_i(\theta)`): list of (parametrized) quantum circuits
+  (a list of :class:`~qiskit.circuit.QuantumCircuit`))
+
+* observables (:math:`H_j`): a list of :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+* grouping: a list of pairs :class:`~qiskit.primitive.Group` of an
+  index of the quantum circuits and an index of the observable.
+  A tuple of two integers can be used as alternative of :class:`~qiskit.primitive.Group`.
+  For example, ``Group(i, j)`` or ``(i, j)`` corresponds to a pair :math:`\psi_i` and :math:`H_j`.
+
+* parameters: a list of parameters of the quantum circuits.
+  (:class:`~qiskit.circuit.parametertable.ParameterView` or
+  a list of :class:`~qiskit.circuit.Parameter`).
+
+* parameter values (:math:`\theta_k`): list of sets of values
+  to be bound to the parameters of the quantum circuits.
+  (list of list of float)
+
+The output is the list of expectation values.
+
+.. math::
+
+    \Braket{\psi_i(\theta_k)|H_j|\psi_i(\theta_k)}
+
+
+The estimator object is expected to be close()d after use or
+accessed inside "with" context
+and the objects are called with parameter values and run options
+(e.g., ``shots`` or number of shots).
+
+Here is an example of how estimator is used.
+
+.. code-block:: python
+
+    from qiskit.circuit.library import RealAmplitudes
+    from qiskit.quantum_info import SparsePauliOp
+
+    psi1 = RealAmplitudes(num_qubits=2, reps=2)
+    psi2 = RealAmplitudes(num_qubits=2, reps=3)
+
+    params1 = psi1.parameters
+    params2 = psi2.parameters
+
+    H1 = SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)])
+    H2 = SparsePauliOp.from_list([("IZ", 1)])
+    H3 = SparsePauliOp.from_list([("ZI", 1), ("ZZ", 1)])
+
+    with Estimator([psi1, psi2], [H1, H2, H3], [params1, params2] as e:
```

Comment:
I think that in this code example is necessity close the Estimator function parenthesis

Reply:
Thanks :+1: